applications/nrf_desktop/doc/battery_charger.rst

```diff
@@ -0,0 +1,56 @@
+| Source Module    | Input Event         | This Module         | Output Event            | Sink Module      |
++==================+=====================+=====================+=========================+==================+
+|                  |                     | ``battery_charger`` | ``battery_state_event`` | :ref:`led_state` |
++------------------+---------------------+                     +-------------------------+------------------+
+| :ref:`usb_state` | ``usb_state_event`` |                     |                         |                  |
++------------------+---------------------+---------------------+-------------------------+------------------+
+
+Configuration
+*************
+
+The module implemented in ``battery_charger.c`` uses the Zephyr :ref:`zephyr:gpio` driver to control and monitor battery charging. For this reason, you should set :option:`CONFIG_GPIO` option.
+
+By default, the module is disabled and the ``CONFIG_DESKTOP_BATTERY_CHARGER_NONE`` option is selected.
+Set the option ``CONFIG_DESKTOP_BATTERY_CHARGER_DISCRETE`` to enable the module.
+
+The following module configuration options are available:
+* ``CONFIG_DESKTOP_BATTERY_CHARGER_CSO_PIN`` - pin to which the charge status output (CSO) from the charger is connected.
+* CSO pin pull:
+
+  * ``CONFIG_DESKTOP_BATTERY_CHARGER_CSO_PULL_NONE`` - disabled (default setting)
+  * ``CONFIG_DESKTOP_BATTERY_CHARGER_CSO_PULL_UP`` - up
+  * ``CONFIG_DESKTOP_BATTERY_CHARGER_CSO_PULL_DOWN`` - down
+
+* ``CONFIG_DESKTOP_BATTERY_CHARGER_ENABLE_PIN`` - pin that enables the battery charging.
+* ``CONFIG_DESKTOP_BATTERY_CHARGER_ENABLE_INVERSED`` - inverse charging enable signal.
+* ``CONFIG_DESKTOP_BATTERY_CHARGER_CSO_FREQ`` - CSO pin state switching frequency for when a charging error occurrs (in Hz).
+
+Implementation details
+**********************
+
+As required by the USB specification, in order to reduce power consumption, the module disables the battery charging when the USB is suspended. In addition, it also disables charging when the USB is disconnected.
+The module enables charging when USB is powered or active.
+
+.. warning:
+  Hardware configuration should enable charging by default (e.g. when pin to enable charging is not configured).
+  Otherwise, it will be impossible to enable charging when the battery is empty.
+
+The module checks the battery state using :c:type:`struct k_delayed_work`.
+The check is based on the edge interrupt count on the CSO pin in a defined period of time (``ERROR_CHECK_TIMEOUT_MS``) and on the CSO pin state while the work is processed.
+
+On a battery state change, the new state is sent using ``battery_state_event``.
+The battery state can have one of the following values:
+* :cpp:enumerator:`BATTERY_STATE_IDLE` - battery is not being charged (CSO pin set high).
+* :cpp:enumerator:`BATTERY_STATE_CHARGING` - battery is being charged (CSO pin set low).
```

Comment:
```suggestion
* :cpp:enumerator:`BATTERY_STATE_CHARGING` - battery is being charged (CSO pin set to `low`).
```

Reply:
can it be logic low and logic high? I think it is more comonly used.